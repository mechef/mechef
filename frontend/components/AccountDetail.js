// @flow

import React from 'react';
import Rx from 'rxjs/Rx';

import Button from './Button';
import { IMAGE_URL } from '../utils/constants';

type Props = {
  account: {
    name?: string,
    kitchenName?: string,
    kitchenDescription?: string,
    firstName?: string,
    lastName?: string,
    phoneNumber?: string,
    email?: string,
    coverPhoto?: string,
    profileImage?: string,
    update: {
      name?: string,
      kitchenName?: string,
      kitchenDescription?: string,
      firstName?: string,
      lastName?: string,
      phoneNumber?: string,
      email?: string,
      coverPhoto?: File,
      profileImage?: File,
    },
  },
  onUpdateAccount: () => Rx.Observable,
}

const AccountDetail = (props: Props) => (
  <div>
    <div className="coverPhoto" />
    <p className="sellerId">{props.account.name}</p>
    <p className="sellerName">{props.account.name}</p>
    <div className="kitchenInfo">
      <div className="name">
        <div className="kitchenName">
          <span className="smallTitle">Kitchen Name</span>
          <span className="contentText">{props.account.kitchenName}</span>
        </div>
        <div className="userName">
          <span className="smallTitle">User Name</span>
          <span className="contentText">{props.account.name}</span>
        </div>
      </div>
      <div className="description">
        <span className="smallTitle">Kitchen Description</span>
        <p className="contentText">{props.account.kitchenDescription || ''}</p>
      </div>
      <div className="name">
        <div className="kitchenName">
          <span className="smallTitle">Phone Number</span>
          <span className="contentText">{props.account.phoneNumber}</span>
        </div>
        <div className="userName">
          <span className="smallTitle">Email Address</span>
          <span className="contentText">{props.account.email}</span>
        </div>
      </div>
      <div className="buttonGroup">
        <Button
          buttonStyle="primary"
          size="small"
          onClick={props.onUpdateAccount}
        >
          UPDATE
        </Button>
      </div>
    </div>
    <div className="additionalInfo">
      <p className="smallTitle">Bank Account</p>
      <div className="infoContent">
        <div className="left">
          <span className="contentText">000 000 0000</span>
        </div>
        <div className="right">
          <Button
            buttonStyle="primary"
            size="small"
            onClick={() => {}}
          >
            EDIT
          </Button>
        </div>
      </div>
    </div>
    <div className="additionalInfo">
      <p className="smallTitle">Password</p>
      <div className="infoContent">
        <div className="left">
          <span className="contentText">******</span>
        </div>
        <div className="right">
          <Button
            buttonStyle="primary"
            size="small"
            onClick={() => {}}
          >
            UPDATE
          </Button>
        </div>
      </div>
    </div>
    <style jsx>
      {`
        .coverPhoto {
          margin-bottom: 25px;
          width: 100%;
          height: 240px;
          background-image: url('${IMAGE_URL}/${props.account.coverPhoto || ''}'), url('../static/pancake.jpg');
          background-size: cover;
          background-position: center;
          position: relative;
        }

        .coverPhoto:after {
          content: '';
          position: absolute;
          top: 200px;
          left: 20px;
          background-image: url('${IMAGE_URL}/${props.account.profileImage || ''}'), url('../static/avatar.jpg');
          background-size: cover;
          background-position: center;
          width: 80px;
          height: 80px;
          border-radius: 40px;
        }

        .sellerId {
          margin-left: 98px;
          margin-bottom: 11px;
          font-size: 24px;
          font-weight: 600;
          line-height: 0.83;
          letter-spacing: 0.6px;
          color: #4a4a4a;
        }

        .sellerName {
          margin-left: 101px;
          margin-bottom: 20px;
          font-size: 14px;
          line-height: 1.43;
          letter-spacing: 0.6px;
          color: #4a4a4a;
        }

        .kitchenInfo {
          display: flex;
          flex-direction: column;
          width: 552px;
          height: 524px;
          padding: 25px 20px;
          margin-left: 21px;
          margin-bottom: 16px;
          border-radius: 4px;
          background-color: #ffffff;
        }

        .name {
          display: flex;
          margin-bottom: 21px;
        }

        .smallTitle {
          font-size: 14px;
          font-weight: 500;
          line-height: 1.14;
          letter-spacing: 0.6px;
          color: #9b9b9b;
        }
        .contentText {
          margin-top: 12px;
          font-size: 16px;
          font-weight: 500;
          line-height: 0.88;
          letter-spacing: 0.7px;
          color: #4a4a4a;
        }
        .kitchenName {
          flex: 1;
          display: flex;
          flex-direction: column;
          height: 70px;
          border-bottom: solid 2px #ececec;
        }

        .userName {
          flex: 1;
          display: flex;
          flex-direction: column;
          height: 70px;
          margin-left: 11px;
          border-bottom: solid 2px #ececec;
        }

        .description {
          display: flex;
          flex-direction: column;
          height: 123px;
          margin-bottom: 20px;
          border-bottom: solid 2px #ececec;
        }

        .buttonGroup {
          display: flex;
          justify-content: flex-end;
          margin-top: auto;
          margin-bottom: auto;
        }

        .additionalInfo {
          display: flex;
          flex-direction: column;
          width: 552px;
          padding: 24px 20px;
          margin-left: 21px;
          margin-bottom: 16px;
          border-radius: 4px;
          background-color: #ffffff;
        }

        .infoContent {
          display: flex;
          justify-content: space-between;
        }

        .left {
          width: 250px;
          height: 70px;
          border-bottom: solid 2px #ececec;
        }
      `}
    </style>
  </div>
);

export default AccountDetail;
