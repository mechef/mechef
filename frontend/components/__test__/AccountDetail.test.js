import renderer from 'react-test-renderer';
import React from 'react';

import AccountDetail from '../AccountDetail';


describe('Pages With Snapshot Testing', () => {
  it('snapshot AccountDetail Component', () => {
    const component = renderer.create(
      <AccountDetail
        account={{
          name: 'bible',
        }}
        onUpdateAccount={() => {}}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
