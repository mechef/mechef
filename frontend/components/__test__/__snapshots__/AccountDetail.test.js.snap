// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages With Snapshot Testing snapshot AccountDetail Component 1`] = `
<div
  className="jsx-72652864"
>
  <div
    className="jsx-72652864 coverPhoto"
  />
  <p
    className="jsx-72652864 sellerId"
  >
    bible
  </p>
  <p
    className="jsx-72652864 sellerName"
  >
    bible
  </p>
  <div
    className="jsx-72652864 kitchenInfo"
  >
    <div
      className="jsx-72652864 name"
    >
      <div
        className="jsx-72652864 kitchenName"
      >
        <span
          className="jsx-72652864 smallTitle"
        />
        <span
          className="jsx-72652864 contentText"
        />
      </div>
      <div
        className="jsx-72652864 userName"
      >
        <span
          className="jsx-72652864 smallTitle"
        />
        <span
          className="jsx-72652864 contentText"
        >
          bible
        </span>
      </div>
    </div>
    <div
      className="jsx-72652864 description"
    >
      <span
        className="jsx-72652864 smallTitle"
      />
      <p
        className="jsx-72652864 contentText"
      >
        
      </p>
    </div>
    <div
      className="jsx-72652864 name"
    >
      <div
        className="jsx-72652864 kitchenName"
      >
        <span
          className="jsx-72652864 smallTitle"
        />
        <span
          className="jsx-72652864 contentText"
        />
      </div>
      <div
        className="jsx-72652864 userName"
      >
        <span
          className="jsx-72652864 smallTitle"
        />
        <span
          className="jsx-72652864 contentText"
        />
      </div>
    </div>
    <div
      className="jsx-72652864 buttonGroup"
    >
      <Button
        buttonStyle="primary"
        onClick={[Function]}
        size="small"
      >
        
      </Button>
    </div>
  </div>
  <div
    className="jsx-72652864 additionalInfo"
  >
    <p
      className="jsx-72652864 smallTitle"
    />
    <div
      className="jsx-72652864 infoContent"
    >
      <div
        className="jsx-72652864 left"
      >
        <span
          className="jsx-72652864 contentText"
        >
          000 000 0000
        </span>
      </div>
      <div
        className="jsx-72652864 right"
      >
        <Button
          buttonStyle="primary"
          onClick={[Function]}
          size="small"
        >
          
        </Button>
      </div>
    </div>
  </div>
  <div
    className="jsx-72652864 additionalInfo"
  >
    <p
      className="jsx-72652864 smallTitle"
    />
    <div
      className="jsx-72652864 infoContent"
    >
      <div
        className="jsx-72652864 left"
      >
        <span
          className="jsx-72652864 contentText"
        >
          ******
        </span>
      </div>
      <div
        className="jsx-72652864 right"
      >
        <Button
          buttonStyle="primary"
          onClick={[Function]}
          size="small"
        >
          
        </Button>
      </div>
    </div>
  </div>
  <JSXStyle
    css=".coverPhoto.__jsx-style-dynamic-selector{margin-bottom:25px;width:100%;height:240px;background-image:url('../static/img/pancake.jpg');background-size:cover;background-position:center;position:relative;}.coverPhoto.__jsx-style-dynamic-selector:after{content:'';position:absolute;top:200px;left:20px;background-image:url('../static/img/avatar.jpg');background-size:cover;background-position:center;width:80px;height:80px;border-radius:40px;}.sellerId.__jsx-style-dynamic-selector{margin-left:116px;margin-bottom:11px;font-size:2.4rem;font-weight:600;line-height:0.83;-webkit-letter-spacing:0.6px;-moz-letter-spacing:0.6px;-ms-letter-spacing:0.6px;letter-spacing:0.6px;color:#4a4a4a;}.sellerName.__jsx-style-dynamic-selector{margin:8px 0 20px 116px;font-size:1.4rem;line-height:1.43;-webkit-letter-spacing:0.6px;-moz-letter-spacing:0.6px;-ms-letter-spacing:0.6px;letter-spacing:0.6px;color:#4a4a4a;}.kitchenInfo.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;max-width:552px;width:calc(100% - 40px);padding:25px 20px;margin-left:20px;margin-right:20px;margin-bottom:16px;border-radius:4px;background-color:#ffffff;box-sizing:border-box;}.name.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;margin-bottom:21px;}.smallTitle.__jsx-style-dynamic-selector{font-size:1.4rem;font-weight:500;line-height:1.14;-webkit-letter-spacing:0.6px;-moz-letter-spacing:0.6px;-ms-letter-spacing:0.6px;letter-spacing:0.6px;color:#9b9b9b;}.contentText.__jsx-style-dynamic-selector{margin-top:12px;font-size:1.6rem;font-weight:500;line-height:0.88;-webkit-letter-spacing:0.7px;-moz-letter-spacing:0.7px;-ms-letter-spacing:0.7px;letter-spacing:0.7px;color:#4a4a4a;}.kitchenName.__jsx-style-dynamic-selector{-webkit-flex:1;-ms-flex:1;flex:1;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;border-bottom:solid 2px #ececec;}.userName.__jsx-style-dynamic-selector{-webkit-flex:1;-ms-flex:1;flex:1;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;height:70px;margin-left:11px;border-bottom:solid 2px #ececec;overflow:auto;}.description.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;height:123px;margin-bottom:20px;border-bottom:solid 2px #ececec;}.buttonGroup.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:end;-webkit-justify-content:flex-end;-ms-flex-pack:end;justify-content:flex-end;margin-top:auto;margin-bottom:auto;}.additionalInfo.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;max-width:552px;width:calc(100% - 40px);padding:25px 20px;margin-left:20px;margin-right:20px;margin-bottom:16px;border-radius:4px;background-color:#ffffff;box-sizing:border-box;}.infoContent.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}.left.__jsx-style-dynamic-selector{width:250px;border-bottom:solid 2px #ececec;}.right.__jsx-style-dynamic-selector{min-width:100px;}@media (max-width:540px){.buttonGroup.__jsx-style-dynamic-selector{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;}.infoContent.__jsx-style-dynamic-selector{-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;}.left.__jsx-style-dynamic-selector{margin-bottom:15px;}infoContent.__jsx-style-dynamic-selector{width:100%;jus;}}"
    dynamic={
      Array [
        "../static/img/pancake.jpg",
        "../static/img/avatar.jpg",
      ]
    }
    styleId="3443884651"
  />
</div>
`;
